shesmu-server-ui/src/simulation.ts

```diff
@@ -0,0 +1,781 @@
+              new DOMParser().parseFromString(olive.diagram, "image/svg+xml")
+                .documentElement
+            ),
+        });
+        const oliveSelection = dropdown(
+          (olive) => [
+            infoForProduces(olive.produces).icon,
+            " ",
+            italic(olive.syntax),
+            " – ",
+            olive.description || "No description.",
+          ],
+          null,
+          oliveState.model,
+          null,
+          ...response.olives
+        );
+        const oliveTabs = tabs(
+          {
+            name: "Overview",
+            contents: oliveState.components.overview,
+          },
+          { name: "Actions", contents: oliveState.components.actions },
+          {
+            name: "Alerts",
+            contents: oliveState.components.alerts,
+            find: alertFindProxy.find,
+          },
+          { name: "Dataflow", contents: oliveState.components.dataflow }
+        );
+        tabList.push({
+          name: "Olive",
+          contents: [oliveSelection, br(), oliveTabs.ui],
+          find: oliveTabs.find,
+        });
+      }
+      if (response.exports?.length) {
+        tabList.push({
+          name: "Exports",
+          contents: response.exports.map((ex) => {
+            switch (ex.type) {
+              case "constant":
+                return [
+                  header(ex.name),
+                  table(
+                    [["Returns", ex.returns]],
+                    ["Position", (x) => x[0]],
+                    ["Type", (x) => x[0]]
+                  ),
+                ];
+              case "function":
+                return [
+                  header(ex.name),
+                  table(
+                    [["Return", ex.returns]].concat(
+                      ex.parameters.map((type, index) => [
+                        `Parameter ${index + 1}`,
+                        type,
+                      ])
+                    ),
+                    ["Position", (x) => x[0]],
+                    ["Type", (x) => x[0]]
+                  ),
+                ];
+              default:
+                return blank();
+            }
+          }),
+        });
+      }
+      if (response.refillers && Object.keys(response.refillers).length) {
+        const refillerState = singleState(
+          ([name, entries]: [string, RefillerRecord[]]) =>
+            entries.length > 0
+              ? renderJsonTable<RefillerRecord>(
+                  name + ".refiller.json",
+                  entries,
+                  ...Object.keys(entries[0])
+                    .sort((a, b) => a.localeCompare(b))
+                    .map(
+                      (name) =>
+                        [name, (row: RefillerRecord) => row[name]] as [
+                          string,
+                          (row: RefillerRecord) => any
+                        ]
+                    )
+                )
+              : "No records provided to refiller."
```

Comment:
Should this be 'No records provided to refill'? or 'No records provided to _the_ refiller'?